### DISCUSSION TITLE
Unit Test Project Type

### DISCUSSION THREAD

**Discussion by victorsolaya** [DynamicsValue/fake-xrm-easy#10] [Q&A] [answered]
Hi Jordi,

First of all, thanks for investing your time in this tool.
I would like to ask some questions that are not quite clear with me at the moment with this version 2.

1. I want to test against Plugins v4.6.2 Framework. Do I need to create a .Net Core Test Project to include your library?
2. Is there any possibility CreateQuery without any assembly type? In your examples you are using createQuery<account>()... However, can I use createQuery without typing the entity name? Or should I use retrieveMultiple in that case?

Thank you!

**Comment by BetimBeja** [comment, marked as answer]
Hello Victor,
there are two versions of FakeXrmEasy:
- v2 that you can use to test your plugins, NuGet package Microsoft.CrmSdk.CoreAssemblies, The test project should target the FULL .Net Framework (ex. v4.6.2)
- v3 that you can use to test your azure functions or integrations based on the new Microsoft.PowerPlatform.Dataverse.Client nuget package.

Regarding Create Query, I am not sure I understood correctly, there are overloads that use Early-Bound entities (where you can omit entity name) or Late-Bound entities where the entity name is required. 

https://github.com/DynamicsValue/fake-xrm-easy-core/blob/8c8bcf1150766c58c4a8415afc00b3f9f11e870d/src/FakeXrmEasy.Core/XrmFakedContext.Queries.cs#L165-L249

**Reply by victorsolaya** [reply]
Hi Betim, 

Thanks for the response.

1. I tried creating a Unit Test Framework, however the library is like not compatible or the number is not correct what it says with the Documentation. I tried with the one that shows in the Releases in Github and I have some issues as well. I will try to replicate again and send here the error.
2. Regarding the Create Query the question cameregarding the fact of retrieving records with that built-in method Create Query without the need of using Early Bound entities. But I see that there is a method for that in the link you shared, so all solved 😊

Thanks for the quick reply!

**Reply by jordimontana82** [reply]
Hi @victorsolaya 

What @BetimBeja  said above is correct. v2.x versions are meant to be used for server side development and you should target them with the full .net framework, v3.x are the ones for .net core.

This is explained [in the docs' Installing section here](https://dynamicsvalue.github.io/fake-xrm-easy-docs/quickstart/installing/). 

Not sure if you already did but, If installing via Nuget Package manager you need to enable the Prerelease tickbox search as these are rc1 packages.

Otherwise, what error did you get?

**Reply by jordimontana82** [reply]
In addition to that: 

Baseline for the FakeXrmEasy.Plugins.v9 -Version 2.x packages is net462. Baseline for FakeXrmEasy.Plugins.v365 -Version 2.x packages is net452.

https://github.com/DynamicsValue/fake-xrm-easy-plugins/blob/2x-dev/src/FakeXrmEasy.Plugins/FakeXrmEasy.Plugins.csproj#L4-L5

**Reply by victorsolaya** [reply]
Hi Jordi. That was the tricky thing. I thought I could do somehow with nuget without downloading the repo.
I could get what you said guys, for that I had to download the repo, then add the nuget source and the clue was that Prerelease checkbox.
Many thanks for your help guys. I will try to document my steps so it can make life easier for some other people.

Thank you!

**Reply by jordimontana82** [reply]
Yeah, no need to download the repo, only the Prerelease bit.

I've updated the Installing section in the docs to make it a bit more obvious :)

**Reply by victorsolaya** [reply]
I see now! I was totally blind, that check was the clue. Many thanks!
